docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx

```diff
@@ -425,14 +425,15 @@ export async function GET() {
 
 #### Using Node.js runtime with local assets
 
-This example uses the Node.js runtime to fetch a local image on the file system and passes it as an `ArrayBuffer` to the `src` attribute of an `<img>` element.
+This example uses the Node.js runtime to fetch a local image on the file system and passes it as an `ArrayBuffer` to the `src` attribute of an `<img>` element. The local asset should be placed relative to the root of your project, instead of the example source code location.
 
 ```jsx filename="app/opengraph-image.js" switcher
 import { ImageResponse } from 'next/og'
+import { resolve } from 'node:path'
 import { readFile } from 'node:fs/promises'
 
 export async function GET() {
-  const logoData = await readFile('./logo.png')
+  const logoData = await readFile(resolve(process.cwd(), './logo.png'))
```

Comment:
```suggestion
  const logoData = await readFile(join(process.cwd(), './logo.png'))
```

Reply:
Thanks for the comments. I'm putting the images in `public` folder, so in my use case, I'm writing `resolve(process.cwd(), './public/logo.png')`, and using `resolve` looks cleaner.

what would you think if I change the file location into `public` folder?

also it comes to my mind that not all files are copied for deploying, would it be better if we suggest making use of `public` folder?

Reply:
> and using resolve looks cleaner

Why is that?

Reply:
> not all files are copied

It will be copied when using `path.join()` since this is statically analyzed by `@vercel/nft`

Reply:
> would it be better if we suggest devs make use of public folder?

Is public copied? I think it ends up in `.next/static` so it would also fail if the output file name doesnt match the one during development.

Reply:
The vercel build logs suggests that the public folder is also copied
<img width="528" alt="image" src="https://github.com/vercel/next.js/assets/3816900/ded0ddca-48d6-411c-a901-57ec1dfa74c8">

Reply:
> Why is that?

hmm, maybe just my personal preference. I'll keep `join` then

Reply:
I'm going to update my PR trying to detail the local file location issue later today.